test/support/test_helpers.rb

```diff
@@ -34,11 +35,17 @@ def setup_page_traffic_data(document_count:)
     commit_index("page-traffic_test")
   end
 
-  def generate_random_example(schema: "help_page", payload: {}, excluded_fields: [])
+  def generate_random_example(schema: "help_page", payload: {}, excluded_fields: [], regenerate_if: ->(_example) { true }, retry_attempts: EXAMPLE_GENERATOR_RETRIES)
     # just in case RandomExample does not generate a type field
+
     payload[:document_type] = schema
-    GovukSchemas::RandomExample
-      .for_schema(notification_schema: schema)
-      .merge_and_validate(payload, excluded_fields)
+    retry_attempts.times do
+      random_example = GovukSchemas::RandomExample
+        .for_schema(notification_schema: schema)
+        .merge_and_validate(payload, excluded_fields)
+
+      return random_example unless regenerate_if.call(random_example)
+    end
+    raise "Could not find example"
```

Comment:
Might need to rescue this. Is it worth creating a specific error class for the message?